diff --git a/node_modules/@mercurjs/framework/dist/utils/index.d.ts b/node_modules/@mercurjs/framework/dist/utils/index.d.ts
index fafc012..be59c99 100644
--- a/node_modules/@mercurjs/framework/dist/utils/index.d.ts
+++ b/node_modules/@mercurjs/framework/dist/utils/index.d.ts
@@ -4,4 +4,5 @@ export * from "./middlewares";
 export * from "./admin";
 export * from "./hosts";
 export * from "./configuration";
+export * from "./store";
 //# sourceMappingURL=index.d.ts.map
\ No newline at end of file
diff --git a/node_modules/@mercurjs/framework/dist/utils/index.js b/node_modules/@mercurjs/framework/dist/utils/index.js
index a22817e..03ea139 100644
--- a/node_modules/@mercurjs/framework/dist/utils/index.js
+++ b/node_modules/@mercurjs/framework/dist/utils/index.js
@@ -20,4 +20,5 @@ __exportStar(require("./middlewares"), exports);
 __exportStar(require("./admin"), exports);
 __exportStar(require("./hosts"), exports);
 __exportStar(require("./configuration"), exports);
+__exportStar(require("./store"), exports);
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/node_modules/@mercurjs/framework/dist/utils/store.d.ts b/node_modules/@mercurjs/framework/dist/utils/store.d.ts
new file mode 100644
index 0000000..f84beea
--- /dev/null
+++ b/node_modules/@mercurjs/framework/dist/utils/store.d.ts
@@ -0,0 +1,3 @@
+import { MedusaContainer } from '@medusajs/framework';
+
+export declare const fetchStoreData: (scope: MedusaContainer) => Promise<any>;
diff --git a/node_modules/@mercurjs/framework/dist/utils/store.js b/node_modules/@mercurjs/framework/dist/utils/store.js
new file mode 100644
index 0000000..a4fd800
--- /dev/null
+++ b/node_modules/@mercurjs/framework/dist/utils/store.js
@@ -0,0 +1,22 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.fetchStoreData = void 0;
+const utils_1 = require("@medusajs/framework/utils");
+async function fetchStoreData(scope) {
+    const query = scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
+    const { data: stores } = await query.graph({
+        entity: 'store',
+        fields: [
+            'id',
+            'name',
+            'default_sales_channel_id',
+            'default_region_id',
+            'default_location_id',
+            'metadata',
+            'supported_currencies.*',
+            'currency.*',
+        ],
+    });
+    return stores?.[0] || null;
+}
+exports.fetchStoreData = fetchStoreData;
